File: dotnet/src/domain/UserUniqueSpecification.cs
namespace DddExample.Domain
{
    /// <summary>
    /// 用户唯一性 Specification 接口
    /// </summary>
    public interface UserUniqueSpecification
    {
        /// <summary>
        /// 用户名是否唯一
        /// </summary>
        bool IsUsernameUnique(string username);

        /// <summary>
        /// 昵称是否唯一
        /// </summary>
        bool IsNicknameUnique(string nickname);

        /// <summary>
        /// 手机号是否唯一
        /// </summary>
        bool IsMobileUnique(string mobile);
    }
}

File: java/src/main/java/com/example/backend/domain/UserUniqueSpecification.java
package com.example.backend.domain;

/**
 * 用户唯一性 Specification 接口
 */
public interface UserUniqueSpecification {

    /**
     * 用户名是否唯一
     */
    boolean isUsernameUnique(String username);

    /**
     * 昵称是否唯一
     */
    boolean isNicknameUnique(String nickname);

    /**
     * 手机号是否唯一
     */
    boolean isMobileUnique(String mobile);
}
